### DISCUSSION TITLE
UI inside node?

### DISCUSSION THREAD

**Discussion by NeoCortex97** [eckig/graph-editor#129] [Q&A]
I have seen the sample with ui inside the nodes. How does this work? Do nodes contain a simple `AnchorPane` or something similar?
I am building a macro plugin for an internal management tool and would design wise avoid a properties panel if possible.

I have seen EMF support in the wiki, but that does not seem like what I need. I only need to add simple controls to the nodes. If persisiting the values of the controls would be possible it would be great, but I could simply add some information to the scripts I generate from the models to restore connections and Positions of the nodes.

**Comment by NeoCortex97** [comment]
We do already have a smaller macro plugin that had code with groovy, but that was very poorly received - to be polite. So now a better system has to be made and since this is simpler to use than blockly inside a `WebView`. I would prefer this Library to any other.

**Comment by eckig** [comment]
Yes, you can place whatever you like inside a `GNode`. The skin is `GNodeSkin` and it consists of a `DraggableBox`, which is more or less just a `StackPane`.
